Source/Core/Common/CommonFuncs.cpp

```diff
@@ -17,29 +17,38 @@ namespace Common
 {
 constexpr size_t BUFFER_SIZE = 256;
 
-// Wrapper function to get last strerror(errno) string.
-// This function might change the error code.
-std::string LastStrerrorString()
+// There are two variants of strerror_r. The XSI version stores the message to the passed-in
+// buffer and returns an int (0 on success). The GNU version returns a pointer to the message,
+// which might have been stored in the passed-in buffer or might be a static string.
+//
+// This function might change the errno value.
+//
+// References:
+// https://www.gnu.org/software/gnulib/manual/html_node/strerror_005fr.html
+// https://pubs.opengroup.org/onlinepubs/9699919799/functions/strerror_r.html
+// https://refspecs.linuxbase.org/LSB_5.0.0/LSB-Core-generic/LSB-Core-generic/baselib-strerror-r.html
+const char* StrErrorWrapper(int error, char* buffer, std::size_t length)
 {
-  char error_message[BUFFER_SIZE];
-
-  // There are two variants of strerror_r. The XSI version stores the message to the passed-in
-  // buffer and returns an int (0 on success). The GNU version returns a pointer to the message,
-  // which might have been stored in the passed-in buffer or might be a static string.
-
   // We check defines in order to figure out variant is in use, and we store the returned value
   // to a variable so that we'll get a compile-time check that our assumption was correct.
```

Comment:
The part about storing the value in a variable to get a type check is no longer true. (For the GNU case, that is.)